lib/secretsManager.js

```diff
@@ -1,77 +1,169 @@
-  let upserted = secrets.find(({ _id }) => _id === secretId)
-
-  if (upserted) {
-    Object.assign(upserted, secret)
-  } else {
-    upserted = { _id: secretId, ...secret }
-    secrets.push(upserted)
+  async encrypt(secrets) {
+    const secretsBytes = aesjs.utils.utf8.toBytes(JSON.stringify(secrets))
+    const aesCtr = new aesjs.ModeOfOperation.ctr(
+      Uint8Array.from(JSON.parse(this.encryptionPassphrase)),
+      new aesjs.Counter(5)
+    )
+    const encryptedBytes = aesCtr.encrypt(secretsBytes)
+    return aesjs.utils.hex.fromBytes(encryptedBytes)
   }
 
-  try {
-    DUMMY_STORE = secrets
-  } catch (err) {
-    console.error('Failed to add new secret to data store', err)
+  async decrypt(encryptedSecrets) {
+    const encryptedBytes = aesjs.utils.hex.toBytes(encryptedSecrets)
+    const aesCtr = new aesjs.ModeOfOperation.ctr(
+      Uint8Array.from(JSON.parse(this.encryptionPassphrase)),
+      new aesjs.Counter(5)
+    )
+    const decryptedBytes = aesCtr.decrypt(encryptedBytes)
+    return JSON.parse(aesjs.utils.utf8.fromBytes(decryptedBytes))
   }
 
-  return upserted
-}
+  async get() {
+    const encrypted = await AsyncStorage.getItem(secretStorageKey)
+    return encrypted ? this.decrypt(encrypted) : []
+  }
+
+  async set(secrets) {
+    return AsyncStorage.setItem(secretStorageKey, await this.encrypt(secrets))
+  }
+
+  /**
+   * Find all secrets, applying a selector if provided.
+   * When provided, all where clause must be fulfilled (logical AND).
+   * By default, returns all secrets
+   * @async
+   * @param {Object} where - hash of where clauses returned secrets must fulfill with
+   * @returns {Array<Object>} list (may be empty) of matching secrets
+   */
+  async find(where = {}) {
+    let secrets = []
 
-/**
- * Remove a secret from the list. Does not fail if it can not be found.
- * @async
- * @param {String} secretId - removed secret id
- * @returns {Object} the removed secret (may be null)
- */
-export async function remove(secretId) {
-  const secrets = await find()
-  const idx = secrets.findIndex(({ _id }) => _id === secretId)
-  let removed = null
-
-  if (idx !== -1) {
-    removed = secrets.splice(idx, 1)[0]
+    try {
+      const props = Object.keys(where)
+      secrets = (await this.get()).filter(secret =>
+        props.every(prop => secret[prop] === where[prop])
+      )
+    } catch (err) {
+      console.error('Failed to read secrets from data store', err)
+    }
+
+    return secrets
   }
 
-  try {
-    DUMMY_STORE = secrets
-  } catch (err) {
-    console.error('Failed to remove secret from data store', err)
+  /**
+   * Add or update a secret to the list.
+   * If not provided, an _id key will be generated
+   * @async
+   * @param {Object} secret - secret to upsert
+   * @returns {Object} the added/updated secret
+   */
+  async upsert(secret) {
+    const secretId = secret._id || uuid()
+    const secrets = await this.find()
+    let upserted = secrets.find(({ _id }) => _id === secretId)
```

Comment:
since we have a find function which accepts parameters, shall we not use that one? just to double check, are we using find with parameters anywhere?

Reply:
I'm not sure, this is again code I just copied verbatim from optic web. I'll give it a look and change if necessary. As per using find with parameters, yes we are. Here is an example: https://github.com/nearform/optic-expo/blob/d83abae38ba52af5baeb0ac74268ce046d1a1c97/context/secrets.js#L37-L37

Reply:
Ok cool, fine then

Reply:
then we're good